ddtrace/appsec/_remoteconfiguration.py

```diff
@@ -89,14 +88,16 @@ def disable_appsec_rc():
 def _add_rules_to_list(features: Mapping[str, Any], feature: str, message: str, ruleset: Dict[str, Any]) -> None:
     rules = features.get(feature, None)
     if rules is not None:
-        try:
-            if ruleset.get(feature) is None:
-                ruleset[feature] = rules
-            else:
+        if ruleset.get(feature) is None:
+            ruleset[feature] = rules
+        else:
+            if isinstance(rules, list) and isinstance(ruleset[feature], list):
                 ruleset[feature] = ruleset[feature] + rules
-            log.debug("Reloading Appsec %s: %s", message, str(rules)[:20])
-        except json.JSONDecodeError:
-            log.error("ERROR Appsec %s: invalid JSON content from remote configuration", message)
+            elif isinstance(rules, dict) and isinstance(ruleset[feature], dict):
+                ruleset[feature] = {**ruleset[feature], **rules}
+            else:
+                log.debug("Invalid type for %s: %s with %s", message, str(type(ruleset[feature])), str(type(rules)))
```

Comment:
Do we have unit tests covering this part of the code?

Reply:
Added better test coverage with different king of merges